docs/developer/architecture-and-design/index.md

```diff
@@ -0,0 +1,9 @@
+# Architecture and Design
```

Comment:
Why not "Architecture decision records"? This would make it easier to find given the Scipp docs and it allows us to add more if need be.

Reply:
I had considered adding this as a subsection. I thought we might have other documentation here as well, not just decision records?

Reply:
ok